### PR TITLE
Merge development into feature branch - Closes #3084

### LINKED ISSUE #3084: Merge Development into feature branch
### Expected behavior

`development` merged into `feature/improve_transactions_processing_efficiency`

### Actual behavior

`development` not merged into `feature/improve_transactions_processing_efficiency`

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

development not merged into feature/improve_transactions_processing_efficiency

### How did I fix it?

By solving conflicts

### How to test it?

Review solved conflicts:

> CONFLICT (content): Merge conflict in framework/test/mocha/unit/modules/chain/submodules/rounds.js
CONFLICT (content): Merge conflict in framework/test/mocha/unit/modules/chain/submodules/blocks/verify.js
CONFLICT (content): Merge conflict in framework/test/mocha/unit/modules/chain/logic/transaction_pool.js
CONFLICT (content): Merge conflict in framework/test/mocha/integration/common.js
CONFLICT (content): Merge conflict in framework/test/mocha/integration/blocks/verify/verify.js
CONFLICT (content): Merge conflict in framework/test/mocha/common/application.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/submodules/rounds.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/submodules/loader.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/submodules/blocks/verify.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/submodules/blocks/process.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/submodules/blocks/chain.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/logic/transaction_pool.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/logic/transaction.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/init_steps/init_modules.js
CONFLICT (content): Merge conflict in framework/src/modules/chain/init_steps/init_logic_structs.js

### Review checklist

* The PR resolves #3084
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by pablitovicente** [comment]
All tests fixes from https://github.com/LiskHQ/lisk/pull/3090 are still valid and behave the same after merging

File: framework/src/modules/chain/submodules/rounds.js [hunk 1]
```diff
@@ -133,25 +133,29 @@ __private.updateRoundInformationWithDelegatesForTransactions = function(
...
-				return self.createRoundInformationWithDelegate(
-					transaction.senderId,
-					round,
-					dependentId,
-					mode,
-					tx
-				);
-			})
-		);
+	return Promise.all(
```
[the remark this answers is not in the document]
Comment (pablitovicente) [reply framework/src/modules/chain/submodules/rounds.js]: Maybe let's open a new issue for this? I didn't change the incoming code only made changes when conflicts happen or rewrites were needed.

[the remark this answers is not in the document]
Comment (shuse2) [reply framework/src/modules/chain/submodules/rounds.js]: ah ok! this was the original code =0 then lets open another issue or fix it with something else

**Comment by pablitovicente** [comment]
The list of tests that should behave the same in both branches is: 

> framework/test/mocha/integration/blocks/chain/apply_block.js
> framework/test/mocha/integration/blocks/chain/delete_last_block_db_trs.js
> framework/test/mocha/integration/blocks/process/on_receive_block.js
> framework/test/mocha/integration/blocks/verify/verify.js
> framework/test/mocha/integration/rounds.js
> framework/test/mocha/integration/transactions/0.0.address_collision.js

**Pull request merged**